### PR TITLE
fix issue of read

### PR DESCRIPTION
**Pull request by xrdavies**
* fix issue of read in #412

### CODE REVIEW

File: client/pool.c [hunk 1]
```diff
@@ -881,14 +881,17 @@ static int receive_data_from_connection(connection_list_element *connection)
...
 	data_size = read(conn_data->connection_descriptor.fd, (uint8_t*)conn_data->data + conn_data->data_size, data_size);
 
-	if(data_size <= 0) {
+	if(data_size < 0) {
 		char message[100] = {0};
 		sprintf(message, "read error : %s", strerror(errno));
 		close_connection(connection, message);
 		return 0;
+	} else if(data_size == 0) {
+		// fixme: read 0
```
Comment (sgaragagghu) [inline client/pool.c]: shouldn't it close the connection?

Comment (xrdavies) [reply client/pool.c]: No need to close, but wait for other data. If close connection, the miners will never connected.

Comment (sgaragagghu) [reply client/pool.c]: maybe the read was interrupted by a signal ?

Comment (xrdavies) [reply client/pool.c]: Not fpga miner, the reporter sent me the screen shot, it's GPU miner.

Comment (sgaragagghu) [reply client/pool.c]: @jonano614 maybe do you know why and when it may send end of file?

Comment (jonano614) [reply client/pool.c]: @sgaragagghu hi. I remember I had some problem with data_size = 0 here, but it was long time ago and I resolved it.

**Pull request merged**